src/AppBundle/Controller/UserController.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use AppBundle\Entity\Category;
+use AppBundle\Entity\Post;
+use AppBundle\Entity\User;
+use AppBundle\Form\Search\SearchUserType;
+use AppBundle\Form\UserType;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Doctrine\Common\Collections\Criteria;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Form\Extension\Core\Type\SearchType;
+
+class UserController extends Controller
+{
+
+    /**
+     * Show user list
+     *
+     * @Route("/user", name="user_list")
+     * @Template()
+     * @Method("GET")
+     */
+    public function indexAction()
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $users = $em->getRepository('AppBundle:User')->findAll();
+        $categories = $em->getRepository('AppBundle:Category')->findAll();
+
+        return $this->render('AppBundle:User:index.html.twig', [
+            'users' => $users,
+            'categories' => $categories,
+        ]);
+    }
+
+    /**
+     * Show user
+     *
+     * @Route("/user/{id}", name="user_show", requirements={"id": "\d+"} )
+     * @Template()
+     * @Method("GET")
+     */
+    public function showAction(Request $request)
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $userRepository = $em->getRepository('AppBundle:User');
+        $categoryRepository = $em->getRepository('AppBundle:Category');
+
+        $user = $userRepository->find($request->get('id'));
```

Comment:
1. id can be as argument in the action. 
2. use param convertor for get user by id from the storage